index.js

```diff
@@ -14,7 +19,10 @@ function removeBomStream() {
 
     buffer = null;
 
-    return removeBom(data);
+    if (isUtf8) {
+      return removeBom.decode(data);
```

Comment:
Can `decode` specify `{ stream: true }` option to avoid needing to wait for the flush?

Reply:
`stream: true` option makes a `TextDecoder` instance use buffering input chunks.
By using this option, [this buffer](https://github.com/gulpjs/remove-bom-stream/blob/52ce320187fa280f97a7321044f1e266b0b4d6eb/index.js#L13) could be replaced to a `TextDecoder` instance. ... No, size of a buffer inside a `TextDecoder` instance cannot be obtained.